### PR TITLE
[SYCLomatic] Update QuickSilver test case.

### PR DESCRIPTION
**Pull request by ShengchenJ**

### CODE REVIEW

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 1]
```diff
@@ -1,148 +1,271 @@
...
+
+
+## Migrating QuickSilver to SYCL
+
+### 1 Prepare the migration
+#### 1.1 Get the source code of QuickSilver
+```sh
+   $ git clone https://github.com/oneapi-src/Velocity-Bench.git
+   $ export QuickSilver_HOME=/path/to/Velocity-Bench/QuickSilver
+   $ cd ${QuickSilver_HOME}/CUDA/src
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion
 
```

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 2]
```diff
@@ -1,148 +1,271 @@
...
+      ├── geomFuncs.cpp
+      ├── mainSift.cpp.dp.cpp
+      └── matching.dp.cpp
+
+   ```
+### 4 Review the migrated source code and fix all `DPCT` warnings
+
+SYCLomatic and [Intel® DPC++ Compatibility Tool](https://www.intel.com/content/www/us/en/developer/tools/oneapi/dpc-compatibility-tool.html) define a list of `DPCT` warnings and embed the warning in migrated source code if need manual effort to check. All the warnings in the migrated code should be reviewed and fixed. For detail of `DPCT` warnings and corresponding fix examples, refer to [Intel® DPC++ Compatibility Tool Developer Guide and Reference](https://www.intel.com/content/www/us/en/develop/documentation/intel-dpcpp-compatibility-tool-user-guide/top/diagnostics-reference.html) or [SYCLomatic doc page](https://oneapi-src.github.io/SYCLomatic/dev_guide/diagnostics-reference.html). 
+
+The icpx supports c++17, so you need to update:
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion
c++17 is required to build SYCL code, so you need to update -std option as follow:
```

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 3]
```diff
@@ -1,148 +1,271 @@
...
+TARGET_0_FLAG_1 = -DHAVE_CUDA -DHAVE_UVM=1 -std=c++11 -O3 ${FLAGS}
+TARGET_0_FLAG_2 = -DHAVE_CUDA -DHAVE_UVM=1 -std=c++11 -O3 ${FLAGS}
+TARGET_0_FLAG_7 = -DHAVE_CUDA -DHAVE_UVM=1 -std=c++11 -O3 ${FLAGS}
+TARGET_0_FLAG_8 = -DHAVE_CUDA -DHAVE_UVM=1 -std=c++11 -O3 ${FLAGS}
+TARGET_0_FLAG_11 = -DHAVE_CUDA -DHAVE_UVM=1 -std=c++11 -O3 ${FLAGS}
+TARGET_0_FLAG_12 = -DHAVE_CUDA -DHAVE_UVM=1 -std=c++11 -O3 ${FLAGS}
+TARGET_0_FLAG_14 = -DHAVE_CUDA -DHAVE_UVM=1 -std=c++11 -O3 ${FLAGS}
+.................
+```
+Some sycl header files depend on the c++17 feature, need to build with c++17.
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion

```

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 4]
```diff
@@ -1,148 +1,271 @@
...
+cycleInit                           10    2.754e+06    2.754e+06    2.754e+06    0.000e+00       100.00
+cycleTracking                       10    7.110e+07    7.110e+07    7.110e+07    0.000e+00       100.00
+cycleTracking_Kernel               107    7.093e+07    7.093e+07    7.093e+07    0.000e+00       100.00
+cycleTracking_MPI                  117    1.751e+05    1.751e+05    1.751e+05    0.000e+00       100.00
+cycleTracking_Test_Done              0    0.000e+00    0.000e+00    0.000e+00    0.000e+00         0.00
+cycleFinalize                       20    3.690e+02    3.690e+02    3.690e+02    0.000e+00       100.00
+Figure Of Merit                  12.67 [Num Mega Segments / Cycle Tracking Time]
+```
+**Note:** 
+* The testing result was running on Intel(R) Core(TM) i7-13700K CPU backend with Intel® oneAPI Base Toolkit(2023.2 version).
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion
* The testing result was collected when run on Intel(R) Core(TM) i7-13700K CPU backend with Intel® oneAPI Base Toolkit(2023.2 version).
```

File: third-party-programs/Velocity-Bench/QuickSilver/README.md (hunk 1 above)
Comment (ShengchenJ) [reply third-party-programs/Velocity-Bench/QuickSilver/README.md]: Done

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 5]
```diff
@@ -1,148 +1,269 @@
...
-
-License and Distribution Information
-------------------------------------
-
-Quicksilver is available [on github](https://github.com/LLNL/Quicksilver)
\ No newline at end of file
+# Migration example: Migrate QuickSilver to SYCL version
+[SYCLomatic](https://github.com/oneapi-src/SYCLomatic) is a project to assist developers in migrating their existing code written in different programming languages to the SYCL* C++ heterogeneous programming model. It is an open source version of Intel® DPC++ Compatibility Tool.
+
+This file lists the detail steps to migrate CUDA version of [QuickSilver](https://github.com/oneapi-src/Velocity-Bench/tree/main/QuickSilver) to SYCL version with SYCLomatic. As follow table summaries the migration environment, software required and so on.
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion
This file lists the detailed steps to migrate CUDA version of [QuickSilver](https://github.com/oneapi-src/Velocity-Bench/tree/main/QuickSilver) to SYCL version with SYCLomatic. As follow table summarizes the migration environment, the software required, and so on.
```

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 6]
```diff
@@ -1,148 +1,269 @@
...
-
-./qs -i ../../Examples/AllScattering/scatteringOnly.inp
-
-License and Distribution Information
-------------------------------------
-
-Quicksilver is available [on github](https://github.com/LLNL/Quicksilver)
\ No newline at end of file
+# Migration example: Migrate QuickSilver to SYCL version
+[SYCLomatic](https://github.com/oneapi-src/SYCLomatic) is a project to assist developers in migrating their existing code written in different programming languages to the SYCL* C++ heterogeneous programming model. It is an open source version of Intel® DPC++ Compatibility Tool.
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion
[SYCLomatic](https://github.com/oneapi-src/SYCLomatic) is a project to assist developers in migrating their existing code written in different programming languages to the SYCL* C++ heterogeneous programming model. It is an open-source version of the Intel® DPC++ Compatibility Tool.
```

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 7]
```diff
@@ -1,148 +1,269 @@
...
+compile_commands.json
+```
+### 3 Migrate the source code and build script
+```sh
+# From the CUDA directory as root directory:
+$ cd ${QuickSilver_HOME}/CUDA
+$ dpct --in-root=. -p=./src/compile_commands.json --out-root=out --gen-build-script --cuda-include-path=/usr/local/cuda/include
+```
+Description of the options: 
+ * `--in-root`: provide input files to specify where to locate the CUDA files that needs migration.
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion
 * `--in-root`: provide input files to specify where to locate the CUDA files that need migration.
```

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 8]
```diff
@@ -1,148 +1,269 @@
...
+ * `-p`: specify compilation database to migrate the whole project.
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion
 * `-p`: specify the compilation database to migrate the whole project.
```

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 9]
```diff
@@ -1,148 +1,269 @@
...
+CrossSection:
+   name: flat
+   A: 0
+   B: 0
+   C: 0
+   D: 0
+   E: 1
+   nuBar: 2.4
+
+WARNING: The enviroment variable SYCL_DEVICE_FILTER is deprecated. Please use ONEAPI_DEVICE_SELECTOR instead.
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion
WARNING: The environment variable SYCL_DEVICE_FILTER is deprecated. Please use ONEAPI_DEVICE_SELECTOR instead.
```

File: third-party-programs/Velocity-Bench/QuickSilver/README.md [hunk 10]
```diff
@@ -1,148 +1,269 @@
...
+[License.txt](https://github.com/oneapi-src/Velocity-Bench/blob/main/QuickSilver/LICENSE.md)
+
+## Reference 
+* Command Line Options of [SYCLomatic](https://oneapi-src.github.io/SYCLomatic/dev_guide/command-line-options-reference.html) or [Intel® DPC++ Compatibility Tool](https://software.intel.com/content/www/us/en/develop/documentation/intel-dpcpp-compatibility-tool-user-guide/top/command-line-options-reference.html)
+* [oneAPI GPU Optimization Guide](https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/)
+* [SYCLomatic project](https://github.com/oneapi-src/SYCLomatic/)
+
+
+## Trademarks information
+Intel, the Intel logo, and other Intel marks are trademarks of Intel Corporation or its subsidiaries.<br>
```
Comment (zhimingwang36) [inline third-party-programs/Velocity-Bench/QuickSilver/README.md]:
```suggestion
Intel, the Intel logo, and other Intel marks are trademarks of Intel Corporation or its subsidiaries.
```

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**